### PR TITLE
Wait for booted website

### LINKED ISSUE #21: Wait for website to finish
At the moment, running `./bin/start` outputs:

> It may take just a moment for everything to spool up.
>
>    http://localhost:3020/

It would be nice if we could detect when the website is actually running.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Depends on #23 being merged first. Closes #21

### CODE REVIEW

File: bin/start [hunk 1]
```diff
@@ -29,6 +29,10 @@ function filterUp() {
     grep -E -v 'is up-to|--remove-orphans'
 }
 
+function testWebsiteConnection() {
+	printf "GET / HTTP/1.1\r\nHost: localhost:3020\r\n\r\n" | nc -w 1 localhost 3020 | grep -c '<body>' || true
```
Comment (iHiD) [inline bin/start:33]: IIRC the `3020` is configurable (via an ENV var maybe? Dunno), so maybe we need to use that config here too?

File: bin/start [hunk 2]
```diff
@@ -73,7 +77,17 @@ else
     docker-compose up -d 2>&1 | filterUp
 
     echo
-    echo "It may take just a moment for everything to spool up."
+    echo -en "Waiting for everything to spool up ..."
+
+    for i in {1..20}
```
Comment (iHiD) [inline bin/start:82]: Suggest 60. Some people might have slow machines.

```suggestion
    for i in {1..60}
```

**Review by iHiD — APPROVED** [review]
Nice work. Approved with suggestions.

File: bin/start (hunk 1 above)
Comment (ErikSchierboom) [reply bin/start:33]: It is configurable, but for that you'd have to edit the port mapping in the `docker-compose-full.yml` file. There is a Docker CLI command to retrieve the port, but that requires the container to already be running, which we could do here, but there are other mentions of the website port in the text that is echoed. I could use some regex magic to extract the data from the compose file, but that is quite ugly.

Comment (ErikSchierboom) [reply bin/start:33]: I'll merge this and then we can decide what to do with this later. I think we could configure the port as an environment variable if we want.

Comment (joshgoebel) [reply bin/start:33]: > It is configurable, but for that you'd have to edit the port mapping in the docker-compose-full.yml file.

No, you'd just edit `stack.yml` and let it update the port mappings.  So you could maybe check `stack.yml` but now you'r switching from Bash to Ruby to parse the yams?  Anyone using a custom port should really be doing it by updating stack.yml, not `compose-full`.

Comment (iHiD) [reply bin/start:33]: @ErikSchierboom Seems like it's probably best to open an issue for this that someone can tackle when they want to change the port and use these scripts?

Comment (ErikSchierboom) [reply bin/start:33]: @iHiD https://github.com/exercism/development-environment/issues/27

**Pull request merged**